daikon-tql/daikon-tql-dsel/src/main/java/org/talend/tqldsel/dseltotql/DselToTqlVisitor.java

```diff
@@ -0,0 +1,286 @@
+            break;
+        case GREATER_OR_EQUAL:
+            comparisonOpEnum = ComparisonOperator.Enum.GET;
+            break;
+        case GREATER_THAN:
+            comparisonOpEnum = ComparisonOperator.Enum.GT;
+            break;
+        case LOWER_OR_EQUAL:
+            comparisonOpEnum = ComparisonOperator.Enum.LET;
+            break;
+        case LOWER_THAN:
+            comparisonOpEnum = ComparisonOperator.Enum.LT;
+            break;
+        default:
+            throw new IllegalStateException("Unsupported comparison operator: " + elNode.getImage());
+        }
+
+        ComparisonExpression comparisonExpression = new ComparisonExpression(
+                new ComparisonOperator(comparisonOpEnum), leftTqlElement, rightTqlElement);
+
+        // Adding it to a new AST
+        AndExpression andExpression = new AndExpression(comparisonExpression);
+        return new OrExpression(andExpression);
+    }
+
+    @Override
+    public TqlElement visitAddOp(ELNode elNode) {
+        throw new IllegalStateException("Unsupported arithmetic operator: " + elNode.getImage());
+    }
+
+    @Override
+    public TqlElement visitMultOp(ELNode elNode) {
+        throw new IllegalStateException("Unsupported arithmetic operator: " + elNode.getImage());
+    }
+
+    @Override
+    public TqlElement visitLogicalOp(ELNode elNode) {
+        switch (elNode.getType()) {
+        case AND:
+            return TqlBuilder.and(buildExpressionsArray(elNode));
+        case OR:
+            return TqlBuilder.or(buildExpressionsArray(elNode));
+        case NOT:
+            return visitNot(elNode);
+        default:
+            throw new IllegalStateException("Unsupported logical operator: " + elNode.getImage());
+        }
+    }
+
+    @Override
+    public TqlElement visitNot(ELNode elNode) {
+        if (isLiteral(elNode.getChild(0).getType().name())) {
+            throw new IllegalArgumentException(
+                    "The parameters of not function cannot be a literal");
+        }
+
+        if (elNode.getChild(0).getType().equals(ELNodeType.HPATH)) {
+            throw new IllegalArgumentException(
+                    "The parameters of not function cannot be an hierarchical path");
+        }
+
+        return TqlBuilder.not((Expression) elNode.getChild(0).accept(this));
+    }
+
+    @Override
+    public TqlElement visitHPath(ELNode elNode) {
+        return new FieldReference(elNode.getImage());
+    }
+
+    @Override
+    public Expression visitVariable(ELNode elNode) {
+        /*
+         * DSEL supports this since V1.1.0 but not TQL
+         */
+        throw new IllegalStateException("Unsupported Variable: " + elNode.getImage());
+    }
+
+    @Override
+    public TqlElement visitFunctionCall(ELNode elNode) {
+        final String fieldName = elNode.getChild(0).getImage();
+
+        switch (elNode.getImage()) {
+        case org.talend.maplang.el.interpreter.impl.function.builtin.IsEmpty.NAME:
```

Comment:
```suggestion
        case IsEmpty.NAME:
```
And same for all case below to be more readable

Reply:
Good idea, thanks, I just did the changes